include/zephyr/device.h

```diff
@@ -419,14 +406,14 @@ typedef int16_t device_handle_t;
 #define DEVICE_DECLARE(name) static const struct device DEVICE_NAME_GET(name)
 
 /**
- * @brief Runtime device dynamic structure (in RAM) per driver instance
+ * @brief Runtime device context (in RAM) per driver instance
  *
  * Fields in this are expected to be default-initialized to zero. The
  * kernel driver infrastructure and driver access functions are
  * responsible for ensuring that any non-zero initialization is done
  * before they are accessed.
  */
-struct device_state {
+struct device_context {
```

Comment:
In my opinion this should be dropped. I think the existing name is OK and I think there should be a high bar to changing the core device model APIs.